numpy/core/src/multiarray/ctors.c

```diff
@@ -1390,10 +1389,12 @@ _array_from_buffer_3118(PyObject *memoryview)
 
         if (!is_ctypes) {
             /* This object has no excuse for a broken PEP3118 buffer */
-            PyErr_SetString(
+            PyErr_Format(
                     PyExc_RuntimeError,
-                    "Item size computed from the PEP 3118 buffer format "
-                    "string does not match the actual item size.");
+                    "Item size %zd for PEP 3118 buffer format "
+                    "string %s does not match the dtype %c item size %zd.",
+                    view->itemsize, view->format, descr->type,
+                    descr->elsize);
```

Comment:
This change would be nice to have in 1.15, especially since we hit this code path more

Reply:
Wrong format for `elsize`, which is `int`.